yaml.go

```diff
@@ -317,3 +317,47 @@ func convertToJSONableObject(yamlObj interface{}, jsonTarget *reflect.Value) (in
 		return yamlObj, nil
 	}
 }
+
+// JSONObjectToYAMLObject converts an in-memory JSON object into a YAML in-memory MapSlice,
+// without going through a byte representation.
+func JSONObjectToYAMLObject(j map[string]interface{}) yaml.MapSlice {
```

Comment:
We already have a method to go from JSonToYaml and vice-versa with a different interface since it takes bytes rather than interface. I'm OK having two interfaces, but now we also have to implementations?

Reply:
`JSONToYAML` is doing something different, and obviously in a different way. We could implement `JSONToYAML` in terms of `json.Unmarshal` and `JSONObjectToYAMLObject`. But that would add one conversion step. I haven't checked, but I expect it to be slower than what we have.

IMO we don't have two interfaces for the same operation, but two distinct operations. `[]byte` is not in-memory YAML. In the same sense an int64 in-memory representation is not a string number.